File: yukicoder/tyama_yukicoder40_2.rb
#!/usr/bin/ruby
gets
e=gets.split.map &:to_i
d=[0,-1,0,1]
c=[]
while d.size<=e.size
c<<x=e[-1]/d[-1]
d.size.times{|i|e[i+e.size-d.size]-=d[i]*x}
e.pop
end
e.pop while e.size>1&&e[-1]==0
p e.size-1
puts e*' '

File: yukicoder/tyama_yukicoder402.py
#!/usr/bin/python
from __future__ import division,print_function
import sys
if sys.version_info[0]<3: input=raw_input

h,w=map(int,input().split())
S=[[ord(c)<46 for c in e]for e in sys.stdin]
for y in range(1,h):
 for x in range(1,w):S[y][x]+=min(S[y-1][x-1],S[y][x-1],S[y-1][x])*S[y][x]
print(-~max(max(e)for e in S)//2)